lint/vale.bzl

```diff
@@ -122,19 +127,22 @@ def vale_action(ctx, executable, srcs, styles, config, stdout, exit_code = None)
 
 # buildifier: disable=function-docstring
 def _vale_aspect_impl(target, ctx):
-    if should_visit(ctx.rule, ctx.attr._rule_kinds, ctx.attr._filegroup_tags):
-        report, exit_code, info = report_files(_MNEMONIC, target, ctx)
-        styles = None
-        if ctx.files._styles:
-            if len(ctx.files._styles) != 1:
-                fail("Only a single directory should be in styles")
-            styles = ctx.files._styles[0]
-            if not styles.is_directory:
-                fail("Styles should be a directory containing installed styles")
-        vale_action(ctx, ctx.executable._vale, ctx.rule.files.srcs, styles, ctx.file._config, report, exit_code)
-        return [info]
-
-    return []
+    if not should_visit(ctx.rule, ctx.attr._rule_kinds, ctx.attr._filegroup_tags):
+        return []
+
+    stdout, report, exit_code, info = report_files(_MNEMONIC, target, ctx)
+    styles = None
+    if ctx.files._styles:
+        if len(ctx.files._styles) != 1:
+            fail("Only a single directory should be in styles")
+        styles = ctx.files._styles[0]
+        if not styles.is_directory:
+            fail("Styles should be a directory containing installed styles")
+    vale_action(ctx, ctx.executable._vale, ctx.rule.files.srcs, styles, ctx.file._config, stdout, exit_code)
+
+    # Run again for machine-readable output, only if rules_lint_report output_group is requested
+    vale_action(ctx, ctx.executable._vale, ctx.rule.files.srcs, styles, ctx.file._config, report, exit_code = "discard", format = "line")
```

Comment:
On Workflows, we'd probably want to _only_ run the machine readable output and not the human readable so we're not running the action twice. In that case we'd loose the exit code output since we're discarding it in this action

Reply:
We might want to do that - but then as discussed, there's nowhere to show the original output in the log, and we figure on CI there's enough parallelism to run double actions?

If we did decide to do that, I'm not sure how to factor this code. There's no way to be conditional on which output is requested. I guess we could just have two exit code outputs, and then the wrapper lint logic chooses which one(s) to read?